internal/cba.go

```diff
@@ -138,6 +138,53 @@ func isClientCertificateEnabled() bool {
 	return strings.ToLower(useClientCert) == "true"
 }
 
+var universePatternToken = "%%UNIVERSE%%"
+
+// getEndpointAndUniverse returns the endpoint for the service as well as the universe
+// domain, taking in to account the various overrides the user may have provided.
+//
+// This method will also select a default endpoint based on MTLS settings, controlled by
+// the GOOGLE_API_USE_MTLS_ENDPOINT environment variable.
+//
+// If the endpoint override is an address (host:port) rather than full base
+// URL (ex. https://...), then the user-provided address will be merged into
+// the default endpoint. For example, WithEndpoint("myhost:8000") and
+// WithDefaultEndpoint("https://foo.com/bar/baz") will return "https://myhost:8080/bar/baz"
+func getEndpointAndUniverse(settings *DialSettings, clientCertSource cert.Source) (string, string, error) {
+	// parameterize the default endpoints with the default universe.
+	defUniverse := getDefaultUniverse(settings)
+	defEndpoint := strings.Replace(settings.DefaultEndpoint, defUniverse, universePatternToken, 1)
+	defMTLSEndpoint := strings.Replace(settings.DefaultMTLSEndpoint, defUniverse, universePatternToken, 1)
+
+	universe := getUniverse(settings)
+	if settings.Endpoint == "" {
+		mtlsMode := getMTLSMode()
+		if mtlsMode == mTLSModeAlways || (clientCertSource != nil && mtlsMode == mTLSModeAuto) {
+			return mergeDefaultEndpointUniverse(defMTLSEndpoint, universe), universe, nil
```

Comment:
If universe domain other than default is configured here, the exception should be "mTLS is not supported in any universe other than googleapis.com".

Reply:
good catch, updated implementation.